### ISSUE TITLE
Export ONNX with V11 produce bad result

### ISSUE THREAD

**Issue by fitoule** [ultralytics/ultralytics#18218]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

I followed the documentation here : https://docs.ultralytics.com/fr/integrations/onnx/#usage
```
from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.pt")

# Export the model to ONNX format
model.export(format="onnx")  # creates 'yolo11n.onnx'

# Load the exported ONNX model
onnx_model = YOLO("yolo11n.onnx")

# Run inference
results = onnx_model("https://ultralytics.com/images/bus.jpg")
```
Here is the result ..
![bus](https://github.com/user-attachments/assets/afd964a9-3798-4ea7-9f6b-ffe6a2f0fd97)

### Environment

yolo checks
Ultralytics 8.3.16 🚀 Python-3.10.14 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 4090, 24215MiB)
Setup complete ✅ (12 CPUs, 62.7 GB RAM, 287.8/456.3 GB disk)

OS                  Linux-6.8.0-49-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.14
Install             pip
RAM                 62.71 GB
Disk                287.8/456.3 GB
CPU                 AMD Ryzen 5 3600 6-Core Processor
CPU count           12
GPU                 NVIDIA GeForce RTX 4090, 24215MiB
GPU count           1
CUDA                11.8

numpy               ✅ 1.23.5>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.0.1+cu118>=1.8.0
torchvision         ✅ 0.15.2+cu118>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
![bus](https://github.com/user-attachments/assets/3d6fdda9-3097-43f9-a83b-e9cd14f9aa53)

pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.8>=2.0.0
torch               ✅ 2.0.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

```
from ultralytics import YOLO

# Load the YOLO11 model
model = YOLO("yolo11n.pt")

# Export the model to ONNX format
model.export(format="onnx")  # creates 'yolo11n.onnx'

# Load the exported ONNX model
onnx_model = YOLO("yolo11n.onnx")

# Run inference
results = onnx_model("https://ultralytics.com/images/bus.jpg")
```

### Additional

I followed the documentation here : https://docs.ultralytics.com/fr/integrations/onnx/#usage

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @fitoule, thank you for bringing this to our attention! 🚀 We appreciate your detailed report and for including your setup details and code example—it’s super helpful.

If this is a 🐛 Bug Report, we’d kindly like to confirm a few things: 

1. Could you verify if this issue persists with the latest version of the `ultralytics` package? You can upgrade to the latest version and its dependencies using the command below:

```bash
pip install -U ultralytics
```

2. Please ensure your environment meets the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Notably, a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) is recommended for compatibility.

3. Could you attach a clear **minimum reproducible example (MRE)** if possible? This will help us reproduce the issue on our side. You’ve already included some code, but any additional details (like ONNX runtime logs, or if this is isolated to specific models) would be valuable. For guidance on creating an MRE, please refer to our [documentation](https://docs.ultralytics.com/help/minimum_reproducible_example/).

Additionally, if you're experiencing concerns specific to your exported ONNX model, you might want to verify the model's export correctness by checking the exported file (`yolo11n.onnx`) for size, shape mismatches, or any errors during the export process.

---

Looking to engage with the Ultralytics community or need real-time assistance? We’ve got you covered:
- Join our [Discord](https://discord.com/invite/ultralytics) for real-time discussions 🎧
- Explore our [Discourse](https://community.ultralytics.com) for structured Q&A and technical discussions.
- Share ideas and learn from the community in our [Subreddit](https://reddit.com/r/Ultralytics).

## Environments

For YOLO tasks, you can also explore these verified environments with all dependencies (CUDA, Python, PyTorch) preinstalled for you:
- **Notebooks** with free GPU:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  
- **Cloud Options**:  
  [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), or [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.
  
---

Our team is actively monitoring reports like this 🛠️. An Ultralytics engineer will review and respond soon to help debug this further. Thank you for your patience and for using Ultralytics! 😊

**Comment by fitoule** [comment]
your yolo checks is not correct "torch ✅ 2.0.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"" I am on unbuntu

**Comment by Y-T-G** [comment]
Can you try latest version?

**Comment by Y-T-G** [comment]
What's the logs from the prediction?

**Comment by fitoule** [comment]
it's good now ! Thank you very much
It's working with the NMS/post processing inside Ultralytic, but I would like to insert my own NMS in the onnx like this post https://github.com/ultralytics/ultralytics/issues/16732#issuecomment-2482078115

**Comment by glenn-jocher** [comment]
@fitoule you're welcome! For custom NMS integration in ONNX, you can refer to our export documentation and adapt the post-processing via ONNX graph modifications. If you need further guidance, feel free to ask.

**Comment by christianepedersen** [comment]
Using yolov10 architecture: I train with ultralytics and then export model to onnx file. Then I use the onnx file for inference and get expected output of shape (1, 6, N obs).

Then when I use yolo11 architecture, train and export to onnx: I get output of ridiculous shape. Like (1, 17, 5148). Very weird. Something is broken with export of yolo11 models to onnx. I think its omitting the output layer or something.

**Comment by Y-T-G** [comment]
@christianepedersen It's correct. YOLOv10 doesn't use NMS, while YOLO11 does. So you need to apply NMS manually when using YOLO11. 

You can check the examples folder to see examples of post-processing.
.https://github.com/ultralytics/ultralytics/tree/main/examples
